### PR TITLE
feat(aspire): full local dev stack on Aspire 13 with multi-runtime debug + emulator bootstrap

### PR DESCRIPTION
**Pull request by arolariu**
## Summary

Replaces the legacy `scripts/dev-local.ts` orchestrator with a **.NET Aspire 13.x AppHost** (`tooling/AppHost`) that drives the entire local dev stack — infra containers (SQL Server, Cosmos vNext, Azurite, Redis) plus all six native dev servers (api/website/exp/cv/docs/status) with full debugger attach, health gates, and a polished Aspire dashboard as the unified dev surface.

A parallel **selfhost mode** (`npm run dev:selfhost`) preserves the existing Docker Compose + Traefik + mkcert stack unchanged for prod-mirror auditing — the two modes coexist.

F5 in VS Code or Visual Studio 2026 launches the entire stack with debuggers automatically attached to .NET, Node (Next.js), and Python (uvicorn).

## Highlights

- **One-command full-stack dev**: `npm run dev` (or F5) brings up infra + 6 services with the dashboard at `https://localhost:17080`.
- **Multi-runtime debugger attach**: `type: "aspire"` launch config auto-attaches per-resource debuggers (`apphost`, `project`, `node`, `python`) — set a breakpoint in C#, TS, or Python and it just hits.
- **Functional readiness gates**: `WaitFor(...)` waits on TDS-accepting-queries (SQL), HTTP `/health` (API), and HTTP `/api/ready` (exp) — not just "container up". Eliminates half-initialized startup failures.
- **Local emulator parity with prod**: Azurite bootstrapped with CORS + `invoices` container + `PublicAccessType.Blob`; Cosmos seeded with `primary` DB + `invoices` / `merchants` containers; SQL `arolariu-sql` database created idempotently. Browser uploads and thumbnail GETs work out of the box.
- **DCP escape hatches documented**: `isProxied: false` for SQL (TDS), uvicorn (HTTP), docusaurus (HTTP) — each with an inline comment explaining the failure mode being avoided.

### AppHost orchestration (Aspire 13.x native APIs)

- New `tooling/AppHost` project + `tooling/AppHost.Tests`
- Native resource declarations: `AddSqlServer`, `AddAzureCosmosDB` (`RunAsPreviewEmulator` Linux vnext), `AddAzureStorage`, `AddRedis` (`.WithoutHttpsCertificate()` for redis:alpine), `AddProject` (api), `AddUvicornApp` (exp), `AddNextJsApp` (website), `AddViteApp` (cv, status), `AddJavaScriptApp` (docs)
- `Constants.cs` — single source for dev credentials, DB names, ports, volume names
- `ExpConfigGenerator` rewrites `sites/exp.arolariu.ro/config.docker.json` with Aspire-allocated localhost endpoints at runtime; restores on graceful shutdown (sidecar `.aspire-bak` for crash recovery)
- `LookupEndpointPort` defensive endpoint-name lookup across Aspire version drift
- `aspire.config.json` committed so `aspire run` works out of the box

### Infra readiness — no more half-started services

- **SQL**: `WithCreationScript` for `arolariu-sql` database, custom `sql-ready` health check that opens a TDS connection to `master` and runs `SELECT 1`. `isProxied: false` to bypass DCP (DCP mishandles TDS — handshake hangs 10+s, pool gets poisoned). `Encrypt=False` works around vpnkit-mangled TLS on Docker Desktop for Windows.
- **Azurite**: `AzuriteBootstrap` helper subscribes to `ResourceReadyEvent`, applies allow-all CORS, and `CreateIfNotExistsAsync("invoices", PublicAccessType.Blob)` with retry+backoff. For pre-existing containers it explicitly upgrades the ACL via `SetAccessPolicyAsync`.
- **Cosmos**: vNext Linux emulator (`RunAsPreviewEmulator`) with `AZURE_COSMOS_EMULATOR_ENABLE_DATA_PLANE_HTTP=true`; `primary` database + `invoices` / `merchants` containers with correct partition keys.
- **API**: `WithHttpHealthCheck("/health")` deep-checks SQL + Cosmos + Storage + exp; replaced exp UrlGroup probe with custom `IConfigProxyClient` ping.
- **exp**: `WithHttpHealthCheck("/api/ready")` config-snapshot readiness (503 on failure).

### Editor & debugger experience

- `.vscode/launch.json`: both Aspire entries use `type: "aspire"` with per-resource `debuggers` block (auto-attaches Node + .NET + Python on F5)
- `.vscode/tasks.json`: dotnet build/publish/watch tasks for VS-Code-aware build flows
- VS 2026 `.esproj` files for each JS site
- Dashboard auto-opens via `DOTNET_DASHBOARD_OPEN_BROWSER=true`
- Fluent UI icons per resource (`Database`, `DatabaseMultiple`, `Storage`, `Memory`, `CodeBlock`, `KeyMultiple`, `Globe`, `PersonAccounts`, `BookOpenGlobe`, `PulseSquare`)

### DCP escape hatches (documented in-place)

Each `isProxied: false` has a comment explaining why DCP would otherwise break:

- **SQL** — DCP mishandles raw TCP/TDS; handshake stalls
- **exp uvicorn** — DCP promotes endpoint URL to https but uvicorn speaks http only
- **docusaurus** — `start` script hardcodes `--port 3100` via CLI flag (ignores `$PORT`); DCP would grab the port first and docusaurus would fail to bind

### OTel observability

- Conditional OTLP exporter on api (.NET) and exp (Python) — wired when `OTEL_EXPORTER_OTLP_ENDPOINT` is set so native processes feed the Aspire dashboard
- exp's Python OTLP exporter explicitly pinned to the HTTP endpoint (21031) — Aspire defaults to gRPC (21030) which the plain-HTTP exporter can't ALPN-negotiate

### npm script swap

- `npm run dev` → `dotnet run --project tooling/AppHost` (Aspire mode, default)
- `npm run dev:selfhost` → full Compose stack (existing behavior, preserved unchanged)
- `scripts/dev-local.ts` + `dev:local*` scripts removed
- F5 in VS Code / VS 2026 launches the AppHost project

### Bundle A — Traefik removal from Aspire mode

Aspire mode runs apps natively on `localhost:PORT` — Traefik adds no value there.
- Deletes `tooling/AppHost/Aspire/TraefikDynamicConfig.cs` (291 lines) + its 3 xUnit tests
- Removes Traefik container declaration, `AddTraefikDynamicConfig(...)` call, `.WaitFor(traefik)` from cv/docs/status, and 4 Traefik-only constants
- Removes stale `infra/Local/Management/traefik/dynamic/.gitignore` (no longer generates `aspire-services.yml`)
- Selfhost-mode Traefik (`infra/Local/Management/traefik/`) unchanged — containerized apps on a Docker network still need a reverse proxy

### Solution & tooling

- `arolariu.sln` removed; `arolariu.slnx` is canonical (includes AppHost projects + `.esproj` files)
- `AGENTS.md` (= `CLAUDE.md`) rewritten to document Aspire vs selfhost as coexisting modes
- `.gitignore` updated: Python `*.egg-info/` artifacts ignored

## Out of scope (explicit non-goals)

- **Selfhost mode is preserved as-is.** Traefik, mkcert, the Compose stacks, `*.localhost` routing — all unchanged. Run `npm run dev:selfhost` for the prod-mirror experience.
- **Friendly resource display names** — would require renaming resources, breaking service-discovery env-var keys (`services__api__http__0`).
- **Health checks for JS dev servers** — Next.js / Vite / SvelteKit / Docusaurus compile on first request, so a probe to `/` would flap during warmup.
- **Aspire env-var injection for exp** (replacing `ExpConfigGenerator` file rewrite) — deferred; current file-rewrite approach works without Python code changes.

## Test plan

- [x] `dotnet build tooling/AppHost` — 0 warnings, 0 errors
- [x] `dotnet build tooling/AppHost.Tests` — 0 warnings, 0 errors
- [x] `dotnet test tooling/AppHost.Tests` — exit 0
- [x] Aspire manifest publisher — succeeds, emits valid JSON for the full 10-resource graph
- [x] Live AppHost smoke: dashboard auto-opens, all 10 resources go green, API serves Swagger UI at `http://localhost:5000/`, website at `https://localhost:3000/`, docs at `http://localhost:3100/`
- [x] Browser scan upload via `uploadScan.ts` — succeeds (Azurite container exists, CORS allows preflight, public-blob policy lets thumbnail GETs through)
- [x] F5 debugger attach — breakpoints hit in C# (api), TS (Next.js website), Python (exp)
- [ ] **For reviewer:** `npm run dev:selfhost` regression check — the existing Compose stack with Traefik should come up unchanged

### CODE REVIEW

**Comment by arolariu** [comment]
# Comprehensive Code Review & DX Enhancement Report

Overall, this is a **stellar** PR. Moving from bespoke `docker-compose` and `.ts` orchestration scripts to the native `.NET Aspire 13.x` model represents a huge leap in developer experience. The multi-runtime debugging integration across Node, .NET, and Python via the Dashboard is brilliant. The telemetry routing, graceful escaping of DCP proxy logic via `isProxied: false` for TDS, and the `AzuriteBootstrap` CORS management are textbook examples of solving hard local-dev pain points.

However, to achieve a **bulletproof, 1:1 production-mirrored Day-0 Developer Experience**, several areas can be further consolidated or refactored:

### 1. Ingress & Subdomains (Routing Parity with Production)
**Current:** Services are accessed via direct localhost ports (e.g., `https://localhost:3000` for Next.js, `http://localhost:5000` for API). Traefik was removed from Aspire mode.
**1:1 Gap:** Production manages ingress through a reverse proxy (Azure Front Door, App Gateway, etc.) hitting subdomains (`api.arolariu.ro`, `arolariu.ro`). Localhost bindings mask CORS issues, SameSite cookie restrictions, and token lifecycle behaviors that only manifest under a real top-level domain.
**Enhancement:** You are already generating `*.localhost` certificates in `postCreate.sh`. Aspire supports custom hosts! You can inject host bindings directly: `.WithEndpoint(..., host: "api.arolariu.localhost")`. This eliminates the need for Traefik locally while enforcing real subdomain testing constraints, validating Cross-Origin boundaries exactly like Production. 

### 2. Day-0 Data Seeding & Zero-Touch Migrations
**Current:** `AppHost` provisions the raw databases (`arolariu-sql` creation script, Cosmos containers, Azurite blob endpoints).
**1:1 Gap:** A fresh developer spins up the environment but encounters a blank schema for SQL and no mock data, requiring manual `dotnet ef database update` or CLI seeding.
**Enhancement:** Adopt the same `builder.Eventing.Subscribe<ResourceReadyEvent>` pattern you masterfully used in `AzuriteBootstrap.cs` to trigger a **Data Seeder Sidecar**. When `sql-ready` resolves, automatically run EF Migrations and seed a standard set of mock merchants/invoices. A developer hitting F5 should log into the UI and immediately see testable data.

### 3. Securing Local Credentials Strategy
**Current:** `Constants.SqlPassword`, `CosmosEmulatorAccountKey` are hardcoded in `tooling/AppHost/Constants.cs`.
**1:1 Gap:** While fine for local emulators, the habit of using static secrets in source code drifts away from Production's Managed Identities / KeyVault posture.
**Enhancement:** Aspire’s parameter references are supported. Instead of falling back to a static constant exclusively, make use of `.NET User Secrets` natively integrated with Aspire. Inject these parameters during startup: 
`builder.AddParameter("sql-password", secret: true)`
The `Constants` fallback is fine, but Aspire helps emulate the “secret injection” pipeline, mimicking how KeyVault delivers these strings via ACR in Prod. 

### 4. Convergence of IaC (Aspire Manifest vs bespoke Bicep)
**Current:** Aspire orchestration is used for local DX, but production deploy relies on `infra/Azure/Bicep`.
**1:1 Gap:** Over time, `AppHost` definitions and Azure Bicep code will drift.
**Enhancement:** Consider adopting `azd` (Azure Developer CLI) integrated with Aspire (`azd infra synth`). Aspire can automatically generate the Bicep template under the hood to ensure that `AppHost` is the *single source of truth* for both local emulation and production provisioning, removing duplicated IaC responsibilities.

### 5. Resilient Health-Checks During Hot-Reload 
**Current:** `WithHttpHealthCheck("/api/ready")` monitors `uvicorn` (Python).
**Enhancement:** `uvicorn --reload` kills and restarts the subprocess. Aspire’s default Health Checks might flag the service as completely deceased during a hot-reload, polluting the Aspire dashboard with red alerts during intensive frontend/python development. Tune the timeout and interval thresholds in `WithHttpHealthCheck(...)` aggressively (e.g., `Delay = 5s, Period = 10s`) inside the AppHost specifically so brief reload blips don’t misreport the service lifecycle.

Once again, this is a phenomenal architecture. Integrating the enhancements above will guarantee that any junior or senior dev joining the team gets a perfect replica of production upon hitting `F5`, complete with seeded data, accurate subdomains, and cloud-native secrets.

File: DEVELOPMENT.md [hunk 1]
```diff
@@ -58,83 +56,90 @@ npm install
...
-4. 🔄 Stops the Docker exp container (frees port 5002 for bare-metal exp)
-5. ⏳ Waits for databases to be healthy, initializes schemas
-6. 🚀 Starts exp, website, and API bare-metal with hot reload
+**What `npm run dev` does:**
+1. ✅ Runs `dotnet run --project tooling/AppHost` — the Aspire AppHost
+2. 🐳 Spawns Aspire-managed containers: Traefik (HTTPS `*.localhost` via mkcert), SQL Server, Cosmos vNext emulator, Redis, Azurite
+3. 🔀 Wires Traefik dynamic routes via `tooling/AppHost/Aspire/TraefikDynamicConfig.cs` (`api.localhost`, `website.localhost`, etc.)
+4. ⏳ Waits for infra health, initializes schemas
+5. 🚀 Starts each app natively with hot reload — dotnet, npm dev scripts, uvicorn
+6. 📊 Exposes the Aspire dashboard at `https://dashboard.localhost` with live OTel traces / metrics / logs
```
[the remark this answers is not in the document]
Comment (arolariu) [reply DEVELOPMENT.md]: Addressed in cc85e5e3: DEVELOPMENT.md Aspire-mode section now drops Traefik / TraefikDynamicConfig refs and documents the actual native-localhost topology with dashboard at https://localhost:17080.

File: DEVELOPMENT.md [hunk 2]
```diff
@@ -58,83 +56,90 @@ npm install
...
 |---------|-----|------|
-| Website | https://localhost:3000 | ✅ Bare-metal, Turbopack hot reload |
-| API | http://localhost:5000 | ✅ Bare-metal, dotnet watch hot reload |
-| exp | http://localhost:5002 | ✅ Bare-metal, uvicorn --reload |
+| Website | https://website.localhost | ✅ Native, Turbopack hot reload |
+| API | https://api.localhost | ✅ Native, .NET Hot Reload |
+| exp | https://exp.localhost | ✅ Native, uvicorn --reload |
+| CV | https://cv.localhost | ✅ Native, Vite HMR |
+| docs | https://docs.localhost | ✅ Native, Docusaurus dev |
+| status | https://status.localhost | ✅ Native, Next.js dev |
```
[the remark this answers is not in the document]
Comment (arolariu) [reply DEVELOPMENT.md]: Addressed in cc85e5e3: DEVELOPMENT.md replaced *.localhost URLs with the actual localhost ports from Program.cs (api 5000, website 3000, exp 5002, cv 4173, docs 3100, status 3002).

File: DEVELOPMENT.md [hunk 3]
```diff
@@ -58,83 +56,90 @@ npm install
...
```
[the remark this answers is not in the document]
Comment (arolariu) [reply DEVELOPMENT.md]: Addressed in cc85e5e3: DEVELOPMENT.md status row updated to 'SvelteKit (AddViteApp)' to match Program.cs.

File: infra/Local/readme.md [hunk 4]
```diff
@@ -2,6 +2,34 @@
...
+- **Reverse proxy**: Traefik (`AddContainer("traefik", ...)`) with bind-mounts to this directory's `traefik/dynamic/` and `certs/`. mkcert-issued cert for `*.localhost`.
+- **Apps as native processes**: exp (Python uvicorn via `AddUvicornApp`), API (.NET via `AddProject`), Website (Next.js via `AddNextJsApp`), CV/status (SvelteKit via `AddViteApp`), docs (Docusaurus via `AddJavaScriptApp`). Hot reload preserved.
+- **Traefik dynamic routes** generated at runtime by `tooling/AppHost/Aspire/TraefikDynamicConfig.cs` into `traefik/dynamic/aspire-services.yml` (gitignored). Routes `https://api.localhost`, `https://website.localhost`, `https://exp.localhost`, `https://cv.localhost`, `https://docs.localhost`, `https://status.localhost`, `https://dashboard.localhost`.
+- **Aspire dashboard**: live OTel traces / metrics / logs at `https://dashboard.localhost`.
+
+In Aspire mode, the `infra/Local/{Storage,Backend,Frontend}/docker-compose.yml` files are NOT used — Aspire spawns its own containers. Only the certs and Traefik dynamic-config dir under `infra/Local/Management/` are referenced (as bind-mounts).
```
[the remark this answers is not in the document]
Comment (arolariu) [reply infra/Local/readme.md]: Addressed in cc85e5e3: infra/Local/readme.md Aspire-mode section rewritten to reflect the current AppHost (no Traefik); selfhost section unchanged.

File: .vscode/README.md [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+|---|---|
+| Selfhost: up | Brings up the legacy containerized stack (`infra/Local/selfhost-start.sh`) |
+| Selfhost: down | Stops the legacy stack |
+| Lint | `npm run lint` |
+| Format | `npm run format` |
+| Test (unit) | `npm run test:unit` |
+| Generate | `npm run generate` |
+
+Aspire mode (F5) does not need any `Infra: up` preLaunchTask — Aspire 13.x's AppHost
+spawns its own infra containers (SQL, Cosmos emulator, Azurite, Redis, Traefik) natively
```
[the remark this answers is not in the document]
Comment (arolariu) [reply .vscode/README.md]: Addressed in cc85e5e3: .vscode/README.md Aspire bullets updated; Traefik dropped, dashboard URL corrected.

File: .vs/README.md [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+
+Open `arolariu.slnx`. AppHost (under `tooling/`) is set as the startup project.
+F5 launches:
+
+1. Aspire 13.x's native infrastructure containers (SQL Server, Cosmos vNext emulator,
+   Azurite, Redis, Traefik) via `tooling/AppHost/Program.cs`
+2. Native processes: exp (Python uvicorn), API (.NET), Website (Next.js),
+   CV (SvelteKit), docs (Docusaurus), status (SvelteKit)
+3. Traefik dynamic config wired for `*.localhost` HTTPS routes
+4. Aspire dashboard at `https://dashboard.localhost`
```
[the remark this answers is not in the document]
Comment (arolariu) [reply .vs/README.md]: Addressed in cc85e5e3: .vs/README.md F5 list updated; Traefik dropped, direct ports + https://localhost:17080 dashboard documented.

File: package.json [hunk 7]
```diff
@@ -35,17 +35,15 @@
...
     "dev:cv": "nx run cv:dev",
     "dev:docs": "nx run docs:dev",
     "dev:exp": "nx run exp:dev",
-    "dev:local": "node scripts/dev-local.ts",
-    "dev:local:frontend": "node scripts/dev-local.ts --frontend",
-    "dev:local:backend": "node scripts/dev-local.ts --backend",
-    "dev:local:infra": "node scripts/dev-local.ts --infra",
-    "dev:local:setup": "node scripts/dev-local.ts --setup-only",
+    "dev:selfhost": "bash infra/Local/selfhost-start.sh",
+    "dev:selfhost:stop": "bash infra/Local/selfhost-stop.sh",
```
[the remark this answers is not in the document]
Comment (arolariu) [reply package.json]: Addressed in 45791849: dev:selfhost now goes through scripts/dev-selfhost.mjs which dispatches to .bat on Windows and .sh on Unix, so 'npm run dev:selfhost' works on both platforms.

File: tooling/AppHost/Constants.cs [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+/// belong here, not configurable values that should come from parameters
+/// or environment variables.
+/// </summary>
+internal static class Constants
+{
+    // ── Dev-only credentials (mirror infra/Local/Storage/docker-compose.yml) ──
+    public const string SqlPassword = "qazWSXedcRFV1234!";
+    public const string RedisPassword = "RedisPassword123!";
+    public const string CosmosEmulatorAccountKey =
+        "C2y6yDjf5/R+ob0N8A7Cgv30VRDJIWEHLM+4QDU5DE2nQ9nDuVTqobD4b8mGGyPMbIZnqyMsEcaGQy67XIw/Jw==";
```
[the remark this answers is not in the document]
Comment (arolariu) [reply tooling/AppHost/Constants.cs]: Addressed in 91ccffc3: SqlPassword/RedisPassword constants removed; Program.cs now resolves them via Aspire parameters backed by configuration (user-secrets / env / appsettings.Development.json). CosmosEmulatorAccountKey renamed to CosmosEmulatorWellKnownKey with XML doc explaining it's the documented public emulator key.

File: tooling/AppHost/Aspire/ExpConfigGenerator.cs [hunk 9]
```diff
@@ -0,0 +1,183 @@
...
+            {
+                state.Ready.Add(evt.Resource);
+                allReady = state.Ready.Count == state.Tracked.Count;
+            }
+
+            if (!allReady)
+                return;
+
+            await WriteAspireConfigAsync(state, connectionStringFactories, ct).ConfigureAwait(false);
+        });
```
[the remark this answers is not in the document]
Comment (arolariu) [reply tooling/AppHost/Aspire/ExpConfigGenerator.cs]: Addressed in 20922526: Added a HasWritten flag inside the existing ReadyLock so WriteAspireConfigAsync runs at most once per AppHost lifecycle even if ResourceReadyEvent re-fires. Reset on StopAsync so a soft restart re-writes.

File: tooling/AppHost/Aspire/ExpConfigGenerator.cs [hunk 10]
```diff
@@ -0,0 +1,183 @@
...
+        {
+            // Restore the cached original content so selfhost mode works next run
+            try
+            {
+                File.WriteAllText(state.ConfigPath, state.OriginalContent);
+            }
+            catch
+            {
+                // best-effort — user can restore manually from config.template.json or git
+            }
```
[the remark this answers is not in the document]
Comment (arolariu) [reply tooling/AppHost/Aspire/ExpConfigGenerator.cs]: Addressed in 20922526: StopAsync now compares the live file content to LastGeneratedContent before restoring; if it diverges the restore is skipped and a warning is logged with the file path, so user edits aren't clobbered.

File: tooling/AppHost.Tests/AppHost.Tests.csproj [hunk 11]
```diff
@@ -0,0 +1,21 @@
...
+    <PackageReference Include="xunit.runner.visualstudio"       Version="3.1.5" />
+    <PackageReference Include="coverlet.collector"              Version="10.0.0">
+      <PrivateAssets>all</PrivateAssets>
+      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
+    </PackageReference>
+  </ItemGroup>
+  <ItemGroup>
+    <ProjectReference Include="..\AppHost\AppHost.csproj" />
+  </ItemGroup>
+</Project>
```
[the remark this answers is not in the document]
Comment (arolariu) [reply tooling/AppHost.Tests/AppHost.Tests.csproj:21]: Addressed in 2c0330f6: Added ExpConfigGeneratorTests covering JSON merge correctness (preserves unrelated keys), idempotent rewrite, and the restore-only-when-unchanged guard.

File: .devcontainer/postCreate.sh [hunk 12]
```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+echo "=== Installing mkcert + generating root CA ==="
+if ! command -v mkcert >/dev/null 2>&1; then
+  MKCERT_VERSION="v1.4.4"
+  curl -sSL "https://github.com/FiloSottile/mkcert/releases/download/${MKCERT_VERSION}/mkcert-${MKCERT_VERSION}-linux-amd64" \
+       -o /tmp/mkcert
+  sudo mv /tmp/mkcert /usr/local/bin/mkcert
+  sudo chmod +x /usr/local/bin/mkcert
```
[the remark this answers is not in the document]
Comment (arolariu) [reply .devcontainer/postCreate.sh]: Addressed in f77a80d5: mkcert download is now sha256-verified (pinned 6d31c65b...) before installation; uses 'install -m 0755' instead of mv+chmod.

File: .devcontainer/postCreate.sh [hunk 13]
```diff
@@ -0,0 +1,47 @@
...
+mkdir -p infra/Local/Management/certs
+cd infra/Local/Management/certs
+if [[ ! -f local-cert.pem ]] || [[ ! -f local-key.pem ]]; then
+  mkcert -key-file local-key.pem -cert-file local-cert.pem "localhost" "*.localhost"
+fi
+cd -
+
+echo "=== Installing Node deps ==="
+npm install
+
```
[the remark this answers is not in the document]
Comment (arolariu) [reply .devcontainer/postCreate.sh]: Addressed in f77a80d5: Removed the redundant 'npm install' from postCreate.sh; updateContentCommand still runs 'npm ci' as the single source of truth.

File: .devcontainer/devcontainer.json [hunk 14]
```diff
@@ -19,36 +16,46 @@
...
+    "4173":  { "label": "CV Site (SvelteKit preview)" },
+    "5000":  { "label": ".NET API" },
+    "5002":  { "label": "exp Config Service" },
+    "5173":  { "label": "CV Site (SvelteKit dev)" },
+    "6006":  { "label": "Storybook Components" },
+    "6379":  { "label": "Redis" },
+    "7007":  { "label": "Storybook Website" },
+    "8080":  { "label": "Traefik dashboard / DocFX" },
+    "8081":  { "label": "Cosmos DB emulator" },
+    "8082":  { "label": "SQL emulator" },
```
[the remark this answers is not in the document]
Comment (arolariu) [reply .devcontainer/devcontainer.json]: Addressed in d1541010: Port 8082 label updated to 'SQL Server' to match the actual mssql container.

File: tooling/AppHost/Program.cs [hunk 15]
```diff
@@ -0,0 +1,297 @@
...
+    // OTEL_EXPORTER_OTLP_ENDPOINT pointing at the gRPC endpoint (21030), which
+    // requires an h2 ALPN handshake the plain-HTTP exporter can't perform —
+    // every metrics / logs export then fails with:
+    //   ssl.SSLError: [SSL] tlsv1 alert no application protocol (_ssl.c:1010)
+    // Explicitly setting protocol + endpoint here picks the right transport.
+    .WithEnvironment("OTEL_EXPORTER_OTLP_PROTOCOL", "http/protobuf")
+    .WithEnvironment("OTEL_EXPORTER_OTLP_ENDPOINT", "https://localhost:21031")
+    .WaitFor(sqlDb)       // waits for sql-ready + database creation
+    .WaitFor(cosmos)
+    .WaitFor(storage)
```
[the remark this answers is not in the document]
Comment (arolariu) [reply tooling/AppHost/Program.cs:191]: Addressed in 20922526: Closed the race: the generator now exposes a ConfigWritten Task (set when the first write completes), and Program.cs uses an async exp.WithEnvironment(...) callback that awaits ConfigWritten before uvicorn is launched, so exp can never read the original Docker-network config.

File: DEVELOPMENT.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (arolariu) [reply DEVELOPMENT.md]: Addressed in cc85e5e3: Same fix as the earlier DEVELOPMENT.md thread — Aspire-mode section rewritten with no Traefik, dashboard at https://localhost:17080, direct localhost ports.

File: infra/Local/readme.md [hunk 16]
```diff
@@ -2,6 +2,34 @@
...
+
+- **Infrastructure**: SQL Server, Cosmos vNext emulator, Azurite, Redis — all spawned by Aspire's native integrations (`AddSqlServer`, `AddAzureCosmosDB().RunAsEmulator()`, etc.) using the same ports/credentials as the Selfhost-mode Compose files so `exp`'s `config.docker.json` keeps working.
+- **Reverse proxy**: Traefik (`AddContainer("traefik", ...)`) with bind-mounts to this directory's `traefik/dynamic/` and `certs/`. mkcert-issued cert for `*.localhost`.
+- **Apps as native processes**: exp (Python uvicorn via `AddUvicornApp`), API (.NET via `AddProject`), Website (Next.js via `AddNextJsApp`), CV/status (SvelteKit via `AddViteApp`), docs (Docusaurus via `AddJavaScriptApp`). Hot reload preserved.
+- **Traefik dynamic routes** generated at runtime by `tooling/AppHost/Aspire/TraefikDynamicConfig.cs` into `traefik/dynamic/aspire-services.yml` (gitignored). Routes `https://api.localhost`, `https://website.localhost`, `https://exp.localhost`, `https://cv.localhost`, `https://docs.localhost`, `https://status.localhost`, `https://dashboard.localhost`.
+- **Aspire dashboard**: live OTel traces / metrics / logs at `https://dashboard.localhost`.
```
[the remark this answers is not in the document]
Comment (arolariu) [reply infra/Local/readme.md]: Addressed in cc85e5e3: Same fix as the earlier infra/Local/readme.md thread — Aspire-mode section rewritten without Traefik refs.

File: .vs/README.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (arolariu) [reply .vs/README.md]: Addressed in cc85e5e3: Same fix as the earlier .vs/README.md thread — startup list updated to match the no-Traefik AppHost.

File: package.json (hunk 7 above)
[the remark this answers is not in the document]
Comment (arolariu) [reply package.json]: Addressed in 45791849: Same fix as the earlier dev:selfhost thread — Node dispatcher (scripts/dev-selfhost.mjs) handles Windows .bat / Unix .sh.

File: .vscode/tasks.json [hunk 17]
```diff
@@ -2,150 +2,80 @@
...
+    },
+    {
+        "label": "watch",
+        "command": "dotnet",
+        "type": "process",
+        "args": [
+            "watch",
+            "run",
+            "--project",
+            "${workspaceFolder}/arolariu.slnx"
```
[the remark this answers is not in the document]
Comment (arolariu) [reply .vscode/tasks.json]: Addressed in fd3ce405: .vscode/tasks.json watch task now points at tooling/AppHost/AppHost.csproj instead of the slnx solution.

File: sites/api.arolariu.ro/src/Core/Domain/General/Extensions/ExpHealthCheck.cs [hunk 18]
```diff
@@ -0,0 +1,53 @@
...
+  public async Task<HealthCheckResult> CheckHealthAsync(
+      HealthCheckContext context,
+      CancellationToken cancellationToken = default)
+  {
+    try
+    {
+      var alive = await configProxy.PingAsync(cancellationToken).ConfigureAwait(false);
+      return alive
+          ? HealthCheckResult.Healthy("exp responded 2xx at /api/ready.")
+          : HealthCheckResult.Unhealthy("exp /api/ready did not return a 2xx status.");
```
[the remark this answers is not in the document]
Comment (arolariu) [reply sites/api.arolariu.ro/src/Core/Domain/General/Extensions/ExpHealthCheck.cs:46]: Addressed in 76db2a39: Added ExpHealthCheckTests covering healthy (Ping returns true), unhealthy (Ping returns false), and exception paths.

File: package.json (hunk 7 above)
[the remark this answers is not in the document]
Comment (arolariu) [reply package.json]: Addressed in 45791849: Same fix as earlier dev:selfhost threads — Node dispatcher works on Windows + Unix.

File: .vscode/tasks.json (hunk 17 above)
[the remark this answers is not in the document]
Comment (arolariu) [reply .vscode/tasks.json]: Addressed in fd3ce405: Same fix as the earlier tasks.json thread — watch now targets AppHost.csproj.

File: .devcontainer/devcontainer.json [hunk 19]
```diff
@@ -19,36 +16,46 @@
...
+    "10000": { "label": "Azurite Blob" },
+    "10001": { "label": "Azurite Queue" },
+    "10002": { "label": "Azurite Table" },
+    "17080": { "label": "Aspire dashboard (HTTP)" },
+    "17081": { "label": "Aspire dashboard (HTTPS)" }
```
[the remark this answers is not in the document]
Comment (arolariu) [reply .devcontainer/devcontainer.json]: Addressed in d1541010: Swapped the 17080/17081 HTTP/HTTPS labels to match the AppHost launch profile (17080=HTTPS, 17081=HTTP).

File: DEVELOPMENT.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (arolariu) [reply DEVELOPMENT.md]: Addressed in cc85e5e3: Same fix as earlier DEVELOPMENT.md threads.

File: infra/Local/readme.md (hunk 16 above)
[the remark this answers is not in the document]
Comment (arolariu) [reply infra/Local/readme.md]: Addressed in cc85e5e3: Same fix as earlier infra/Local/readme.md threads.

File: .vs/README.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (arolariu) [reply .vs/README.md]: Addressed in cc85e5e3: Same fix as earlier .vs/README.md threads.

File: .vscode/README.md [hunk 20]
```diff
@@ -0,0 +1,44 @@
...
+via `tooling/AppHost/Program.cs`.
```
[the remark this answers is not in the document]
Comment (arolariu) [reply .vscode/README.md]: Addressed in cc85e5e3: Same fix as earlier .vscode/README.md threads.

File: tooling/AppHost.Tests/AppHost.Tests.csproj [hunk 21]
```diff
@@ -0,0 +1,21 @@
...
+    <PackageReference Include="xunit"                           Version="2.9.3" />
+    <PackageReference Include="xunit.runner.visualstudio"       Version="3.1.5" />
+    <PackageReference Include="coverlet.collector"              Version="10.0.0">
+      <PrivateAssets>all</PrivateAssets>
+      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
+    </PackageReference>
+  </ItemGroup>
+  <ItemGroup>
+    <ProjectReference Include="..\AppHost\AppHost.csproj" />
+  </ItemGroup>
```
[the remark this answers is not in the document]
Comment (arolariu) [reply tooling/AppHost.Tests/AppHost.Tests.csproj:20]: Addressed in 2c0330f6: Same fix as the earlier AppHost.Tests thread — ExpConfigGeneratorTests added.
